src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -3220,196 +3220,141 @@ void java_lang_reflect_AccessibleObject::set_override(oop reflect, jboolean valu
+int java_lang_reflect_Method::_annotation_default_offset;
 
-#define CONSTRUCTOR_FIELDS_DO(macro) \
-  macro(_clazz_offset,          k, vmSymbols::clazz_name(),          class_signature,       false); \
-  macro(_parameterTypes_offset, k, vmSymbols::parameterTypes_name(), class_array_signature, false); \
-  macro(_exceptionTypes_offset, k, vmSymbols::exceptionTypes_name(), class_array_signature, false); \
-  macro(_slot_offset,           k, vmSymbols::slot_name(),           int_signature,         false); \
-  macro(_modifiers_offset,      k, vmSymbols::modifiers_name(),      int_signature,         false); \
-  macro(_signature_offset,             k, vmSymbols::signature_name(),             string_signature,     false); \
-  macro(_annotations_offset,           k, vmSymbols::annotations_name(),           byte_array_signature, false); \
-  macro(_parameter_annotations_offset, k, vmSymbols::parameter_annotations_name(), byte_array_signature, false);
+#define METHOD_FIELDS_DO(macro) \
+  macro(_name_offset,           k, vmSymbols::name_name(),           string_signature,      false); \
+  macro(_returnType_offset,     k, vmSymbols::returnType_name(),     class_signature,       false); \
+  macro(_annotation_default_offset,    k, vmSymbols::annotation_default_name(),    byte_array_signature, false);
 
-void java_lang_reflect_Constructor::compute_offsets() {
-  InstanceKlass* k = vmClasses::reflect_Constructor_klass();
-  CONSTRUCTOR_FIELDS_DO(FIELD_COMPUTE_OFFSET);
+void java_lang_reflect_Method::compute_offsets() {
+  InstanceKlass* k = vmClasses::reflect_Method_klass();
+  METHOD_FIELDS_DO(FIELD_COMPUTE_OFFSET);
 }
 
 #if INCLUDE_CDS
-void java_lang_reflect_Constructor::serialize_offsets(SerializeClosure* f) {
-  CONSTRUCTOR_FIELDS_DO(FIELD_SERIALIZE_OFFSET);
+void java_lang_reflect_Method::serialize_offsets(SerializeClosure* f) {
+  METHOD_FIELDS_DO(FIELD_SERIALIZE_OFFSET);
 }
 #endif
 
-Handle java_lang_reflect_Constructor::create(TRAPS) {
+Handle java_lang_reflect_Method::create(TRAPS) {
   assert(Universe::is_fully_initialized(), "Need to find another solution to the reflection problem");
-  Symbol* name = vmSymbols::java_lang_reflect_Constructor();
-  Klass* k = SystemDictionary::resolve_or_fail(name, true, CHECK_NH);
-  InstanceKlass* ik = InstanceKlass::cast(k);
-  // Ensure it is initialized
-  ik->initialize(CHECK_NH);
-  return ik->allocate_instance_handle(THREAD);
-}
-
-oop java_lang_reflect_Constructor::clazz(oop reflect) {
-  return reflect->obj_field(_clazz_offset);
-}
-
-void java_lang_reflect_Constructor::set_clazz(oop reflect, oop value) {
-   reflect->obj_field_put(_clazz_offset, value);
-}
-
-oop java_lang_reflect_Constructor::parameter_types(oop constructor) {
-  return constructor->obj_field(_parameterTypes_offset);
-}
-
-void java_lang_reflect_Constructor::set_parameter_types(oop constructor, oop value) {
-  constructor->obj_field_put(_parameterTypes_offset, value);
+  Klass* klass = vmClasses::reflect_Method_klass();
```

Comment:
This also does not need a cast, vmClasses::reflect_Method_klass() returns an InstanceKlass.